### PR TITLE
ci: Update op-e2e

### PR DESCRIPTION
**Pull request by trianglesphere**
**Description**

This fixes go based CI in several ways
1. Set test.parallel flag so it matches the actual amount of CPUs request
2. Add more calls to t.Parallel in op-e2e
3. Reduce the amount of blocks in TestBigL2Txs to reduce the test time.

CI Time for op-e2e goes from 16 min -> 4:30 min.

**Metadata**

- Fixes CLI-3762

### CODE REVIEW

File: op-e2e/actions/l2_batcher_test.go [hunk 1]
```diff
@@ -428,7 +428,7 @@ func TestBigL2Txs(gt *testing.T) {
 	}
 
 	// build many L2 blocks filled to the brim with large txs of random data
-	for i := 0; i < 100; i++ {
+	for i := 0; i < 40; i++ {
```
Comment (trianglesphere) [inline op-e2e/actions/l2_batcher_test.go:431]: @protolambda I reduced the amount of blocks here b/c this test was slow + resource intensive

**Review by protolambda — APPROVED** [review]
amazing

**Review by ajsutton — APPROVED** [review]
LGTM. Nice work.  I've been running all the e2e tests locally a fair bit recently and IntelliJ runs them in parallel which has been quite reliable. I'd be surprised if there isn't some intermittency that turns up so we'll need to keep an eye out and fix it when it does, but it shouldn't be frequent from what I've seen running locally.

**Comment by trianglesphere** [comment]
> LGTM. Nice work. I've been running all the e2e tests locally a fair bit recently and IntelliJ runs them in parallel which has been quite reliable. I'd be surprised if there isn't some intermittency that turns up so we'll need to keep an eye out and fix it when it does, but it shouldn't be frequent from what I've seen running locally.

So far with the changes I've been making it's been more reliable this time (though I will watch our for flakes). I think the big improvement is specifying the parallelism. We may have been hitting this CircleCI issue where it doesn't appear to be setting the `proc/cpu_count` value correctly: https://circleci.com/docs/configuration-reference/#resourceclass

**Review by mslipper — APPROVED** [review]

**Review by zhwrd — APPROVED** [review]

**Pull request merged**